### DISCUSSION TITLE
Running Airflow with custom log handlers in addition to default Airflow loggers

### DISCUSSION THREAD

**Discussion by dcodoshawks** [apache/airflow#35534] [Q&A]
We have a collection of Python processes that we are in the process of migrating to being run from Airflow. In these processes, we use a custom log configuration that defines a format we like as well as a rotating file we log to. We define this logging configuration in `logging.ini` file and then when these scripts first start we load this config. Having a single file for these logs allows simple downstream monitoring such as uploading to AWS CloudWatch.

Now that we are starting to run these tasks from Airflow Python operators, we are facing issues unifying it with Airflow's logging. If we do not initialize our custom logger, then Airflow's logging works properly but it does not have our format and does not log to our custom file. On the flip side, if we initialize our logger then it logs to our custom file but not to Airflow's log locations, so the UI shows no logs.

Is there a simple way to combine these two log configs without messing up Airflow's logging?

**Comment by hbc-acai** [comment]
Is having two handlers an option, i.e. one to log into airflow's log directory and another to AWS? 

Also, you can move airflow logs to AWS by using [AWSLogsHook](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/cloud-watch-task-handlers.html ).

**Reply by dcodoshawks** [reply]
@hbc-acai Having two handlers is an option, but I wasn't sure of the best way to merge our existing handlers with the default Airflow handlers. Are there any best practices?

**Reply by hbc-acai** [reply]
I think when you call logging.fileConfig with disable_existing_loggers=False, it should merge your settings in logging.ini with the airflow log settings.
